pittapi/news.py

```diff
@@ -107,13 +110,39 @@ def _get_page_articles(
     return page_articles
 
 
+def get_categories() -> list[str]:
+    global CATEGORY_URL_NAME_MAP
+    if not CATEGORY_URL_NAME_MAP:
+        CATEGORY_URL_NAME_MAP = _scrape_categories()
+    return list(CATEGORY_URL_NAME_MAP.keys())
+
+
+def get_topics() -> list[str]:
+    global TOPIC_ID_MAP
+    if not TOPIC_ID_MAP:
+        TOPIC_ID_MAP = _scrape_topics()
+    return list(TOPIC_ID_MAP.keys())
```

Comment:
Fun suggestion, use `functools.cache`!

```suggestion
from functools import cache

@cache
def get_categories() -> list[str]:
    category_map = _scrape_categories()
    return list(category_map.keys())

@cache
def get_topics() -> list[str]:
    topic_id_map = _scrape_topics()
    return list(topic_id_map.keys())
```